test/test-pipe-set-non-blocking.c

```diff
@@ -16,53 +16,63 @@
 #include "uv.h"
 #include "task.h"
 
-#ifdef _WIN32
-
-TEST_IMPL(pipe_set_non_blocking) {
-  RETURN_SKIP("Test not implemented on Windows.");
-}
-
-#else  /* !_WIN32 */
-
-#include <errno.h>
-#include <string.h>
-#include <sys/socket.h>
-#include <sys/types.h>
-#include <sys/un.h>
-#include <unistd.h>
-
 struct thread_ctx {
   uv_barrier_t barrier;
-  int fd;
+  uv_os_fd_t fd;
 };
 
 static void thread_main(void* arg) {
   struct thread_ctx* ctx;
+  uv_fs_t req;
+  uv_buf_t bufs[1];
   char buf[4096];
   ssize_t n;
+  int uv_errno;
+
+  bufs[0] = uv_buf_init(buf, sizeof(buf));
 
   ctx = arg;
   uv_barrier_wait(&ctx->barrier);
 
-  do
-    n = read(ctx->fd, buf, sizeof(buf));
-  while (n > 0 || (n == -1 && errno == EINTR));
+  do {
+    uv_sleep(0.25); /* make sure we are forcing the writer to block */
+    uv_errno = uv_fs_read(NULL, &req, ctx->fd, bufs, 1, -1, NULL);
+    n = req.result;
+    uv_fs_req_cleanup(&req);
+  } while (n > 0 || (n == -1 && uv_errno == UV_EINTR));
 
+#ifdef _WIN32
+  ASSERT(n == UV_EOF);
+#else
   ASSERT(n == 0);
+#endif
 }
 
+
+#ifdef _WIN32
+static void write_cb(uv_write_t* req, int status) {
+  ASSERT(status == 0);
+  req->handle = NULL; /* signal completion of write_cb */
+}
+#endif
+
+
+
 TEST_IMPL(pipe_set_non_blocking) {
   struct thread_ctx ctx;
   uv_pipe_t pipe_handle;
   uv_thread_t thread;
   size_t nwritten;
   char data[4096];
   uv_buf_t buf;
-  int fd[2];
+  uv_os_fd_t fd[2];
   int n;
+#ifdef _WIN32
+  uv_write_t write_req;
+#endif
 
   ASSERT(0 == uv_pipe_init(uv_default_loop(), &pipe_handle, 0));
-  ASSERT(0 == socketpair(AF_UNIX, SOCK_STREAM, 0, fd));
+  ASSERT(0 == uv_pipe(fd, 0, 0));
```

Comment:
shouldn't we use `uv_socketpair()` here?

Reply:
I could make it conditional on the OS. On Windows, this needs to call `uv_pipe`, since the rest of the test is expecting a uv_os_fd_t.

Reply:
I noticed about this because the test was failing locally and then thought about the unix sockets vs pipes thing

Reply:
That was my fault. I changed the API of `uv_pipe` after doing my testing (passing `int fd[2]` instead of `int read, int write`), and got the ordering backwards here (so it was trying to read/write the wrong ends of the pipe).